hal/armv8m/nrf/91/nrf91.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * nRF91 basic peripherals control functions
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Damian Loewnau
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "nrf91.h"
+
+#include "../../../cpu.h"
+#include "../../armv8m.h"
+#include "../../../../include/errno.h"
+
+
+static struct {
+	volatile u32 *scb;
+	volatile u32 *power;
+	volatile u32 *clock;
+	volatile u32 *gpio;
+	u32 cpuclk;
+	spinlock_t pltctlSp;
+} nrf91_common;
+
+
+/* clang-format off */
+enum { power_tasks_constlat = 30, power_tasks_lowpwr, power_inten = 192, power_intenset, power_intenclr, power_status = 272};
+
+
+enum { clock_tasks_hfclkstart = 0, clock_inten = 192, clock_intenset, clock_intenclr, clock_hfclkrun = 258, clock_hfclkstat };
+
+
+enum { gpio_out = 1, gpio_outset, gpio_outclr, gpio_in, gpio_dir, gpio_dirsetout, gpio_dirsetin, gpio_cnf = 128 };
+
+
+enum { syst_csr = 4, syst_rvr, syst_cvr, syst_calib };
+
+
+enum { fpu_cpacr = 34, fpu_fpccr = 141, fpu_fpcar, fpu_fpdscr };
+/* clang-format on */
+
+
+/* platformctl syscall */
+
+
+/* TODO: add platformctl implementation */
+int hal_platformctl(void *ptr)
+{
+	return -ENOSYS;
+}
+
+
+void _nrf91_platformInit(void)
```

Comment:
I'd name this function `_hal_platformInit`, like on `armv7a` and `sparc` targets. This way there'll be no need for an `#ifdef` in `hal_cpuInit`.

Reply:
Applied, this and the other suggestions. Moreover, I've looked through the whole my code in kernel once again.